code/src/components/Form.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+
+const Form = ({newMessage, onNewMessage, onSubmitForm}) => {
+  return (
+
+    <form onSubmit={onSubmitForm} className='form-container'>
+
+      <label htmlFor='newMessage' className='form-label'>Add your happy thought: </label>
+      <input
+        id='newMessage'
+        type='text'
+        value={newMessage}
+        onChange={onNewMessage}
+        className='form-input'
+      />
+      <button type='submit' className='form-btn'>SEND HAPPY THOUGHT</button>
+    </form>
+  )
+}
+
+export default Form
```

Comment:
I think as a tip you could make is so that your site will refresh automatically when you press submit on the form! Good job of having such a clean component.

Reply:
Not refresh, but re-render the list with re-fetched data. There should be never such thing as refreshing the single page application.